Tests/WitsmlExplorer.Api.Tests/Workers/AnalyzeGapWorkerTests.cs

```diff
@@ -0,0 +1,253 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Serilog;
+using Witsml;
+using Witsml.Data;
+using Witsml.Data.Curves;
+using Witsml.Extensions;
+using Witsml.ServiceReference;
+using WitsmlExplorer.Api.Jobs;
+using WitsmlExplorer.Api.Models;
+using WitsmlExplorer.Api.Models.Reports;
+using WitsmlExplorer.Api.Services;
+using WitsmlExplorer.Api.Workers;
+using Xunit;
+
+namespace WitsmlExplorer.Api.Tests.Workers;
+
+/// <summary>
+/// Create gap analysis tests.
+/// </summary>
+public class AnalyzeGapWorkerTests
+{
+    private const string LogUid = "logUid";
+    private const string WellUid = "wellUid";
+    private const string WellboreUid = "wellboreUid";
+    private const string CurveIndex = "Depth";
+    private const string MnemonicList = "Depth,BPOS,SPM1";
+   
+    private const string DepthDataRow1 = "51,88,100";
+    private const string DepthDataRow2 = "52,88,100";
+    private const string DepthDataRow3 = "53,88,100";
+    
+    private const string DepthDataWithGapsRow1 = "51,88,100";
+    private const string DepthDataWithGapsRow2 = "52,,";
+    private const string DepthDataWithGapsRow3 = "53,92,";
+    private const string DepthDataWithGapsRow4 = "54,94,130";
+   
+    private const string TimeDataRow1 = "2023-04-19T00:00:00Z,88,100";
+    private const string TimeDataRow2 = "2023-04-19T00:00:01Z,88,100";
+    private const string TimeDataRow3 = "2023-04-19T00:00:02Z,88,100";
+    
+    private const string TimeDataWithGapsRow1 = "2023-04-19T00:00:00Z,88,100";
+    private const string TimeDataWithGapsRow2 = "2023-04-19T00:00:01Z,,";
```

Comment:
Log data rows where only the index is present are not possible in practice (at least for Sitecom servers afaik).

Reply:
fixed